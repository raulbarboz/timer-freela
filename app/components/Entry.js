import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { addCompany } from '../actions/entry';

function Entry(){
    const dispatch = useDispatch();
    const [company, setCompany ] = useState('');

    const handlerInput = (e) => {
        const { value } = e.target;
        setCompany(value)
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        if(company){
            dispatch(addCompany(company))
            setCompany('')
        }
    }
    // useEffect to call function
    // After any update will send information to parent
    useEffect(() => {
       
    })

    return(
        <div className="entry">
            <form>
                <input 
                value={company}
                type="text" 
                name="company" 
                placeholder="Cadastrar nova empresa"
                onChange = { 
                    e => handlerInput(e)
                } />
                <button
                type="submit"
                onClick={
                    e => handlerSubmit(e)
                }
                >
                Enviar
                </button>
            </form>
        </div>
    )
}

export default Entry;