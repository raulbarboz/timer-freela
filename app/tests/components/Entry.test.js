import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Entry from '../../components/Entry';

test('should render Entry correctly', () => {
    const wrapper = shallow(<Entry />);
    expect(toJSON(wrapper)).toMatchSnapshot()
})