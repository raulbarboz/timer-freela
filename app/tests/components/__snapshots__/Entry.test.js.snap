// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Entry correctly 1`] = `
<div
  className="entry"
>
  <input
    name="from"
    onChange={[Function]}
    placeholder="DDD DE ORIGEM"
    type="text"
    value=""
  />
  <input
    name="to"
    onChange={[Function]}
    placeholder="DDD DE DESTINO"
    type="text"
    value=""
  />
  <input
    name="minutes"
    onChange={[Function]}
    placeholder="MINUTOS"
    type="text"
    value=""
  />
  <select
    name="plan"
    onChange={[Function]}
  >
    <option
      value="30"
    >
      Fale + 30
    </option>
    <option
      value="60"
    >
      Fale + 60
    </option>
    <option
      value="120"
    >
      Fale + 120
    </option>
  </select>
</div>
`;
